PowerGlove_BLE_MIDI/powerglove_ble_midi.py

```diff
@@ -0,0 +1,126 @@
+"""
+Power Glove BLE MIDI with Feather Sense nRF52849
+Sends MIDI CC values based on finger flex sensors and accelerometer
+"""
+import time
+import board
+import adafruit_ble
+from adafruit_ble.advertising.standard import ProvideServicesAdvertisement
+import adafruit_ble_midi
+import adafruit_midi
+from adafruit_midi.control_change import ControlChange
+# from adafruit_midi.note_on import NoteOn
+# from adafruit_midi.pitch_bend import PitchBend
+import adafruit_lsm6ds  # accelerometer
+import simpleio
+from analogio import AnalogIn
+
+i2c = board.I2C()
+sense_accel = adafruit_lsm6ds.LSM6DS33(i2c)
+
+analog_in_thumb = AnalogIn(board.A3)
+analog_in_index = AnalogIn(board.A2)
+analog_in_middle = AnalogIn(board.A1)
+analog_in_ring = AnalogIn(board.A0)
+
+# Pick your MIDI CC numbers here
+cc_x_num = 7  # volume
+cc_y_num = 70  # unassigned
+cc_thumb_num = 71  # unassigned
+cc_index_num = 75  # unassigned
+cc_middle_num = 76  # unassigned
+cc_ring_num = 77  # unassigned
+
+midi_channel = 1  # pick your midi out channel here
+
+# Use default HID descriptor
+midi_service = adafruit_ble_midi.MIDIService()
+advertisement = ProvideServicesAdvertisement(midi_service)
+
+ble = adafruit_ble.BLERadio()
+if ble.connected:
+    for c in ble.connections:
+        c.disconnect()
+
+midi = adafruit_midi.MIDI(midi_out=midi_service, out_channel=midi_channel - 1)
+
+print("advertising")
+ble.start_advertising(advertisement)
+
+# reads an analog pin and returns value remapped to out range, e.g., 0-127
+def get_flex_cc(sensor, low_in, high_in, min_out, max_out):
+    flex_raw = sensor.value
+    flex_cc = simpleio.map_range(flex_raw, low_in, high_in, min_out, max_out)
+    flex_cc = int(flex_cc)
+    return flex_cc
+
+
+debug = False  # set debug mode True to test raw values, set False to run BLE MIDI
+
+while True:
+    if debug:
+        accel_data = sense_accel.acceleration  # get accelerometer reading
+        accel_x = accel_data[0]
+        accel_y = accel_data[1]
+        accel_z = accel_data[2]
+
+        print(
+            "x:{} y:{} z:{} thumb:{} index:{} middle:{} ring:{}".format(
+                accel_x,
+                accel_y,
+                accel_x,
+                analog_in_thumb.value,
+                analog_in_index.value,
+                analog_in_middle.value,
+                analog_in_ring.value,
+            )
+        )
+        time.sleep(0.2)
+
+    else:
+        print("Waiting for connection")
+        while not ble.connected:
+            pass
+        print("Connected")
+        while ble.connected:
+
```

Comment:
Kinda weird to have a blank line as the first after the previous ends in :

Reply:
right on, will fix.